stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -1638,15 +1638,29 @@ julia> c = normalize(a, 1)
 
 julia> norm(c, 1)
 1.0
+
+julia> a = [[1 2 4] ; [1 2 4]]
+2×3 Array{Int64,2}:
+ 1  2  4
+ 1  2  4
+
+julia> norm(a)
+6.48074069840786
+
+julia> normalize(a)
+2×3 Array{Float64,2}:
+ 0.154303  0.308607  0.617213
+ 0.154303  0.308607  0.617213
+
 ```
 """
-function normalize(v::AbstractVector, p::Real = 2)
-    nrm = norm(v, p)
-    if !isempty(v)
-        vv = copy_oftype(v, typeof(v[1]/nrm))
-        return __normalize!(vv, nrm)
+function normalize(a::AbstractArray, p::Real = 2)
+    nrm = norm(a, p)
+    if !isempty(a)
+        aa = copy_oftype(a, typeof(first(a)/nrm))
```

Comment:
Cool that you found this bug! Can you add a test with a e.g. OffaxisArray that does not have index 1?

Reply:
added a test case based on a testcase I found in test/offset.jl

```
julia> A = OffsetArray([-1,0], (-2,))
2-element OffsetArray{Int64,1,Array{Int64,1}} with indices -1:0:
 -1
  0
julia> Arsc = reshape(A, :, 1)
2×1 reshape(::OffsetArray{Int64,1,Array{Int64,1}}, 2, 1) with eltype Int64:
 -1
  0
julia> Arsc[1,1] = 5
5
julia> first(A)
5
julia> A[1]
ERROR: BoundsError: attempt to access 2-element OffsetArray{Int64,1,Array{Int64,1}} with indices -1:0 at index [1]
```
Works with normalize now:
```
julia> normalize(A)
2-element OffsetArray{Float64,1,Array{Float64,1}} with indices -1:0:
 1.0
 0.0
```

Reply:
Yeah that is a good test case!